### PR TITLE
Story1 display collection

### PR DESCRIPTION
**Pull request by Freschwamm**

### CODE REVIEW

File: index.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+$db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+
+$query = $db->prepare('SELECT * FROM `pokemon-card`;');
+
+$query->execute();
+
+$allPokemon = $query->fetchAll();
+
+require_once 'functions.php';
```
Comment (norayna) [inline index.php]: put this at the top before PDO stuff (this is just convention)

File: functions.php [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+            '<section class="card-stats">' .
+            "<h2> Name: " . $pokemon['name'] . "</h2>".
+            '<img src="' . $pokemon['image'] . '" alt="' . $pokemon['name'] . '">' .
+            "<p> Cost: " .$pokemon['cost'] . "</p>" .
+            "<p> Rarity: ". $pokemon['rarity'] . "<p/>" . "</section>";
+    }
+    {
+        if ($allPokemon === [['name' => 'lizard', 'image' => "lizard",
+                'cost' => 'lizard', 'rarity'=> 'lizard']]) {
+            return "That's not a pokemon, it's a lizard";
```
Comment (norayna) [inline functions.php]: this is saying that if the function receives this exact incorrect input, then it returns an error message. it doesn't take care of all possible incorrect inputs, it just chooses the specific one that happens to be in the unit test.

in the future, someone might edit your code and, for example, accidentally pass the wrong array into the function. your function would end up throwing an error that would be displayed on the page plainly for users to see.

you need to handle all possible invalid arrays (valid arrays should have all the keys you need to create the html).

Comment (norayna) [reply functions.php]: also unfortunately you need to put a more informative (boring) error message

File: functions.php [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+function pokemonCardContainer(array $allPokemon): string {
+    $pokemonNames = "";
+    foreach ($allPokemon as $pokemon) {
+        $pokemonNames .=
+            '<section class="card-stats">' .
+            "<h2> Name: " . $pokemon['name'] . "</h2>".
+            '<img src="' . $pokemon['image'] . '" alt="' . $pokemon['name'] . '">' .
+            "<p> Cost: " .$pokemon['cost'] . "</p>" .
+            "<p> Rarity: ". $pokemon['rarity'] . "<p/>" . "</section>";
```
Comment (norayna) [inline functions.php]: this html is different than what the test expects so the unit test for this function fails

**Review by norayna — CHANGES_REQUESTED** [review]
overall very good!

File: index.php (hunk 1 above)
Comment (Freschwamm) [reply index.php]: Done

File: functions.php (hunk 3 above)
Comment (Freschwamm) [reply functions.php]: fixed

File: functions.php (hunk 2 above)
Comment (Freschwamm) [reply functions.php]: fixed

File: functions.php [hunk 4]
```diff
@@ -2,24 +2,25 @@
...
-            '<img src="' . $pokemon['image'] . '" alt="' . $pokemon['name'] . '">' .
-            "<p> Cost: " .$pokemon['cost'] . "</p>" .
-            "<p> Rarity: ". $pokemon['rarity'] . "<p/>" . "</section>";
+function pokemonCardContainer(array $allPokemon): string { $pokemonNames = "";
+        foreach ($allPokemon as $pokemon) {
+            if (!(array_key_exists('name', $pokemon) &&
+                array_key_exists('image', $pokemon) &&
+                array_key_exists('cost', $pokemon) &&
+                array_key_exists('rarity', $pokemon))) {
+                return "PLease enter correct information";
```
Comment (norayna) [inline functions.php]: really sorry but you've capitalised PLease weird here

**Review by norayna — CHANGES_REQUESTED** [review]
sorry

**Review by norayna — APPROVED** [review]

**Pull request merged**